jetson/science_bridge/src/__main__.py

```diff
@@ -0,0 +1,391 @@
+
+    def thermistor_handler(self, msg, thermistor_struct):
+        # msg format: <"$THERMISTOR,temp0,temp1,temp2">
+        try:
+            arr = msg.split(",")
+            thermistor_struct.temp0 = float(arr[1])
+            thermistor_struct.temp1 = float(arr[2])
+            thermistor_struct.temp2 = float(arr[3])
+        except Exception as e:
+            print("thermistor exception:", e)
+            pass
+        # parse the thermistor UART msg
+        # adding relevant error handling
+        # set the struct variables
+
+    def triad_handler(self, m, triad_struct):
+        # msg format: <"$TRIAD,d0_1,d0_2, .... , d2_6">
+        try:
+            m.split(',')
+            arr = [s.strip().strip('\x00') for s in m.split(',')]
+            struct_variables = ["d0_1", "d0_2", "d0_3", "d0_4", "d0_5", "d0_6",
+                                "d1_1", "d1_2", "d1_3", "d1_4", "d1_5", "d1_6",
+                                "d2_1", "d2_2", "d2_3", "d2_4", "d2_5", "d2_6"]
+            count = 1
+            for var in struct_variables:
+                if (not (count >= len(arr))):
+                    pass
+                    setattr(triad_struct, var, 0xFFFF & ((np.uint8(arr[count+1]) << 8) | (np.uint8(arr[count]))))
+                else:
+                    setattr(triad_struct, var, 0)
+                count += 2
+        except Exception as e:
+            print("triad exception:", e)
+            pass
+
+        # parse the spectral UART msg
+        # add in relevant error handling
+        # set the struct variables
+
+    def txt_handler(self, msg, struct):
+        '''
+        Prints info messages revieved from nucleo to screen
+        '''
+        print(msg)
+
+    def carousel_handler(self, msg, carousel_struct):
+        # msg format: <"$CAROUSEL,position>
+        try:
+            arr = msg.split(",")
+            carousel_struct.position = np.int8(arr[1])
+        except Exception as e:
+            print(e)
+
+    def heater_state_handler(self, msg, struct):
+        # msg format: <"$HEATER,device,enabled">
+        try:
+            arr = msg.split(",")
+            # Send back the heater and the state
+            struct.device = int(arr[1])
+            struct.enabled = bool(int(arr[2]))
+        except Exception as e:
+            print(e)
+
+    def heater_shutoff_handler(self, msg, struct):
+        # msg format: <"$auto_shutoff,device,enabled">
+        try:
+            arr = msg.split(",")
+            enabled = False
+            if (arr[1] == "1"):
+                enabled = True
+            print(enabled)
+            struct.auto_shutdown_enabled = enabled
+        except Exception as e:
+            print(e)
+
+    def heater_transmit(self, channel, msg):
+        # Upon Receiving a heater on/off command, send a command for the appropriate mosfet
+        print("Heater cmd callback")
+        struct = Heater.decode(msg)
+        message = "$Mosfet,{device},{enable},1111111"
```

Comment:
i really hate the padding convention..

Reply:
also why does it have 1s in this string if you add them in an if statement later in the code?

Reply:
fixed